### PR TITLE
REST API: Job Listings

### PR DESCRIPTION
**Pull request by pgk**
Fixes #

#### Changes proposed in this Pull Request:

* Enables job-listings as a custom post type under /wp/v2/job-listings
* Adds a custom nested object field to the above (fields)

Notes:

The custom nested object is modelled as a Mixtape Model.

#### Testing instructions:

tested with node 7. Run `npm install`, then `./node_modules/.bin/mixtape build`

With WPJM rest api enabled you can:

* Access wp/v2/job-listings
* Create a new job listing
* Update an existing job listing
* Delete a job listing

### CODE REVIEW

File: includes/rest-api/class-wp-job-manager-registrable-job-listings.php [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+
+		return $this->model_factory->create( $data, array(
+			'deserialize' => true,
+		) );
+	}
+
+	/**
+	 * Our Reader.
+	 *
+	 * @param mixed 		  $data Data.
```
Comment (jom) [inline includes/rest-api/class-wp-job-manager-registrable-job-listings.php]: Tabs

**Review by jom — CHANGES_REQUESTED** [review]
Overall, it looks really good. Few minor changes requested. Did some testing and works for me.

File: mixtape.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
```
[the remark this answers is not in the document]
Comment (pgk) [reply mixtape.json]: Good catch! 💯

**Comment by pgk** [comment]
@jom sorry, this got buried under the mass of PRs, changes made :)

**Review by jom — APPROVED** [review]

**Comment by Nisha9219** [comment]
how to access categories plz help me

**Pull request merged**